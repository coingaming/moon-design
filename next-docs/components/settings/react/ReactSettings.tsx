import { useState } from 'react';
import {
  LogoSportsbetShort,
  LogoBitcasinoShort,
  LogoMoonDesignShort,
  LogoBetaddaShort,
} from '@heathmont/moon-assets';
import useFeatureFlags from '../../../utils/useFeatureFlags';
import { useDocsTheme, DocsBrands } from '../../themes/DocsThemeProvider';
import useThemeTW from '../../themes/useThemesTW';
import ReactElixirSwitcher from '../common/ReactElixirSwitcher';
import SettingsButton from '../common/SettingsButton';
import DarkLightModeSwitcher from './DarkLightModeSwitch';
import RTLModeSwitch from './RTLModeSwitch';

type LogosKeys = keyof typeof Logos;

interface Props {
  toggleReactAndElixir: () => void;
  isReactEnabled: boolean;
}

const Logos = {
  sportsbet: <LogoSportsbetShort fontSize="1rem" />,
  bitcasino: <LogoBitcasinoShort fontSize="1rem" />,
  moonDesign: <LogoMoonDesignShort fontSize="1rem" />,
  betadda: <LogoBetaddaShort fontSize="1rem" />,
};

const ReactSettings: React.FC<Props> = ({
  toggleReactAndElixir,
  isReactEnabled,
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const toggle = () => setIsOpened(!isOpened);

  const { toggleColorScheme, setBrand, themeKeys } = useDocsTheme();

  const [isRtlEnabled, setIsRtl] = useState(false);

  const { isFeatureElixirEnabled } = useFeatureFlags();

  const toggleRtl = () => {
    const htmlTag = document && document?.getElementsByTagName('html')[0];
    setIsRtl(!isRtlEnabled);
    const dirAttr = htmlTag.getAttribute('dir');

    if (!dirAttr) {
      htmlTag.setAttribute('dir', 'rtl');
    } else {
      htmlTag.removeAttribute('dir');
    }
  };

  const filteredThemeKeys = themeKeys.filter((themeKey: string) =>
    ['sportsbet', 'bitcasino', 'moonDesign', 'betadda'].includes(themeKey)
  );

  const {
    setBrand: setBrandTW,
    toggleMode: toggleModeTW,
    getMode,
  } = useThemeTW();

  const switchBrandHandler = (themeKey: LogosKeys) => {
    setBrand(themeKey as DocsBrands);
    setBrandTW(themeKey);
  };

  const switchModeHandler = () => {
    toggleColorScheme();
    toggleModeTW();
  };
  return (
    <>
      <SettingsButton onClick={toggle} isRtlEnabled={isRtlEnabled} />
      {isOpened && (
        <>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-16 z-50`}
          >
            {filteredThemeKeys.map((themeKey: string) => {
              const Logo = Logos[themeKey as LogosKeys] || Logos.moonDesign;
              return (
                <button
                  key={themeKey}
                  type="button"
                  onClick={() => switchBrandHandler(themeKey as LogosKeys)}
                  className={`${
                    isRtlEnabled ? `mr-4` : `ml-4`
                  } p-2 rounded-full text-black bg-white hover:bg-gray-200 inline-flex items-center justify-center shadow-md`}
                >
                  {Logo}
                </button>
              );
            })}
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-28 z-50`}
          >
            <DarkLightModeSwitcher
              toggle={switchModeHandler}
              isEnabled={getMode() === 'dark'}
              isRtlEnabled={isRtlEnabled}
            />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-40 z-50`}
          >
            <RTLModeSwitch toggle={toggleRtl} isEnabled={isRtlEnabled} />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-52 z-50`}
          >
            {isFeatureElixirEnabled ? (
              <ReactElixirSwitcher
                toggle={toggleReactAndElixir}
                isReactEnabled={isReactEnabled}
              />
            ) : null}
          </div>
        </>
      )}
    </>
  );
};

export default ReactSettings;
