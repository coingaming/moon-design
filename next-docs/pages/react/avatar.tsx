import React from 'react';

import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageAvatar() {
  return (
    <>
      <ThemeProvider theme={sportsbetDark}>
        <h1>Avatar</h1>
      </ThemeProvider>
    </>
  );
}
