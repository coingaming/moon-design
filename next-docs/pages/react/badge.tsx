import React from 'react';
import Head from 'next/head';

import { Badge } from '@heathmont/moon-core';
import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageBadge() {
  return (
    <>
      <h1>Badge</h1>
      <p>Small count and labeling component.</p>
      <Badge>Active</Badge>
      <Badge color="goten.100" backgroundColor="piccolo.100">
        Active
      </Badge>
    </>
  );
}
