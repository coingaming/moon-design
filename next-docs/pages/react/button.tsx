import React from 'react';
import Head from 'next/head';

import { Button } from '@heathmont/moon-components';
import PreviewSwitch from '../../components/codePreview/Switch';
import Preview from '../../components/codePreview/Preview';

export default function PageBadge() {
  return (
    <>
      <section className="mt-6">
        <h1 className="text-5xl font-semibold">Button</h1>
        <p className="text-lg mt-4">
          Buttons are calls-to-action used to prompt users. They encourage users
          to interact with us in multiple ways throughout our galaxy, based on
          what the label of the button indicates.
        </p>

        <p className="text-lg mt-4">
          Buttons are clickable elements with label text that describe the
          action that will happen when the users interact with it.
        </p>
      </section>
      <section className="mt-6">
        <Preview
          title="Primary"
          preview={<Button variant="primary">Active</Button>}
          code={`import { Button } from '@heathmont/moon-components';

<Button variant="primary">Active</Button>`
}
        />
      </section>
    </>
  );
}
