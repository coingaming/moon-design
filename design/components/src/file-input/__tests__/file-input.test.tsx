import * as React from 'react';
import { create } from 'react-test-renderer';

import { FileInput } from '..';

describe('File Input', () => {
  test('renders a file input with default placeholder', () => {
    const fileInput = create(<FileInput id="fileInput1" />);

    expect(fileInput).toMatchSnapshot();
  });

  test('renders a file input with custom placeholder', () => {
    const fileInput = create(
      <FileInput id="fileInput2" placeholder="Choose a photoâ€¦" />
    );

    expect(fileInput).toMatchSnapshot();
  });

  test('renders a file input with custom label', () => {
    const fileInput = create(
      <FileInput id="fileInput3" label="ID, passport, etc." />
    );

    expect(fileInput).toMatchSnapshot();
  });

  test('renders a file input with custom label and placeholder', () => {
    const fileInput = create(
      <FileInput
        id="fileInput4"
        label="Upload photo"
        placeholder="ID, passport, etc."
      />
    );

    expect(fileInput).toMatchSnapshot();
  });

  test('renders as error', () => {
    const fileInput = create(<FileInput id="fileInput5" error />);

    expect(fileInput).toMatchSnapshot();
  });
});
