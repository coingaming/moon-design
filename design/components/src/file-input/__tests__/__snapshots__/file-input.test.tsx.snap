// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Input renders a file input with custom label 1`] = `
.emotion-1 {
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  color: #fff;
  background-color: transparent;
  border: 1px solid #415662;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  padding-right: 2.625rem;
  background-position: right 0.5rem center;
  background-repeat: no-repeat;
  background-size: 1.25rem;
  position: relative;
  display: block;
  cursor: pointer;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 18' xmlns='http://www.w3.org/2000/svg' color='%238697a2' class='css-uk6cul'%3e%3cpath d='M10.902 18H9.098V8.423l-2.78 2.69L5 9.838 10 5l5 4.838-1.318 1.275-2.78-2.69V18zM1.81 1.861V6H0V0h20v6h-1.84V1.861H1.81z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e");
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: #304854;
}

.emotion-1:disabled,
.emotion-1:disabled + label {
  color: #304854;
}

.emotion-1:focus-within {
  border-color: #8697a2;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<label
  className="emotion-1"
>
  <span>
    Upload photo
  </span>
  <input
    className="emotion-0"
    type="file"
  />
</label>
`;

exports[`File Input renders a file input with custom label and placeholder 1`] = `
.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-3 {
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  color: #fff;
  background-color: transparent;
  border: 1px solid #415662;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  padding-right: 2.625rem;
  background-position: right 0.5rem center;
  background-repeat: no-repeat;
  background-size: 1.25rem;
  position: relative;
  display: block;
  cursor: pointer;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 18' xmlns='http://www.w3.org/2000/svg' color='%238697a2' class='css-uk6cul'%3e%3cpath d='M10.902 18H9.098V8.423l-2.78 2.69L5 9.838 10 5l5 4.838-1.318 1.275-2.78-2.69V18zM1.81 1.861V6H0V0h20v6h-1.84V1.861H1.81z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e");
  padding-top: 1.3125rem;
  padding-bottom: 0.4375rem;
}

.emotion-3:disabled {
  cursor: not-allowed;
  border-color: #304854;
}

.emotion-3:disabled,
.emotion-3:disabled + label {
  color: #304854;
}

.emotion-3:focus-within {
  border-color: #8697a2;
}

.emotion-1 {
  position: absolute;
  left: 0.875rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  color: #8697a2;
  pointer-events: none;
  -webkit-transform: translateY(-90%) scale(0.75);
  -ms-transform: translateY(-90%) scale(0.75);
  transform: translateY(-90%) scale(0.75);
}

<label
  className="emotion-3"
>
  <span>
    Upload photo
  </span>
  <span
    className="emotion-1"
  >
    <span
      className="emotion-0"
    >
      . 
    </span>
    ID, passport, etc.
  </span>
  <input
    className="emotion-0"
    type="file"
  />
</label>
`;

exports[`File Input renders a file input with custom placeholder 1`] = `
.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-3 {
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  color: #fff;
  background-color: transparent;
  border: 1px solid #415662;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  padding-right: 2.625rem;
  background-position: right 0.5rem center;
  background-repeat: no-repeat;
  background-size: 1.25rem;
  position: relative;
  display: block;
  cursor: pointer;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 18' xmlns='http://www.w3.org/2000/svg' color='%238697a2' class='css-uk6cul'%3e%3cpath d='M10.902 18H9.098V8.423l-2.78 2.69L5 9.838 10 5l5 4.838-1.318 1.275-2.78-2.69V18zM1.81 1.861V6H0V0h20v6h-1.84V1.861H1.81z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e");
  padding-top: 1.3125rem;
  padding-bottom: 0.4375rem;
}

.emotion-3:disabled {
  cursor: not-allowed;
  border-color: #304854;
}

.emotion-3:disabled,
.emotion-3:disabled + label {
  color: #304854;
}

.emotion-3:focus-within {
  border-color: #8697a2;
}

.emotion-1 {
  position: absolute;
  left: 0.875rem;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  color: #8697a2;
  pointer-events: none;
  -webkit-transform: translateY(-90%) scale(0.75);
  -ms-transform: translateY(-90%) scale(0.75);
  transform: translateY(-90%) scale(0.75);
}

<label
  className="emotion-3"
>
  <span>
    Choose file
  </span>
  <span
    className="emotion-1"
  >
    <span
      className="emotion-0"
    >
      . 
    </span>
    ID, passport, etc.
  </span>
  <input
    className="emotion-0"
    type="file"
  />
</label>
`;

exports[`File Input renders a file input with default label 1`] = `
.emotion-1 {
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  color: #fff;
  background-color: transparent;
  border: 1px solid #415662;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  padding-right: 2.625rem;
  background-position: right 0.5rem center;
  background-repeat: no-repeat;
  background-size: 1.25rem;
  position: relative;
  display: block;
  cursor: pointer;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 18' xmlns='http://www.w3.org/2000/svg' color='%238697a2' class='css-uk6cul'%3e%3cpath d='M10.902 18H9.098V8.423l-2.78 2.69L5 9.838 10 5l5 4.838-1.318 1.275-2.78-2.69V18zM1.81 1.861V6H0V0h20v6h-1.84V1.861H1.81z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e");
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: #304854;
}

.emotion-1:disabled,
.emotion-1:disabled + label {
  color: #304854;
}

.emotion-1:focus-within {
  border-color: #8697a2;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<label
  className="emotion-1"
>
  <span>
    Choose file
  </span>
  <input
    className="emotion-0"
    type="file"
  />
</label>
`;

exports[`File Input renders as error 1`] = `
.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-1 {
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  color: #fff;
  background-color: transparent;
  border: 1px solid #415662;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  padding-right: 2.625rem;
  background-position: right 0.5rem center;
  background-repeat: no-repeat;
  background-size: 1.25rem;
  position: relative;
  display: block;
  cursor: pointer;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 18' xmlns='http://www.w3.org/2000/svg' color='%238697a2' class='css-uk6cul'%3e%3cpath d='M10.902 18H9.098V8.423l-2.78 2.69L5 9.838 10 5l5 4.838-1.318 1.275-2.78-2.69V18zM1.81 1.861V6H0V0h20v6h-1.84V1.861H1.81z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e");
  border-color: #ff4e64;
}

.emotion-1:disabled {
  cursor: not-allowed;
  border-color: #304854;
}

.emotion-1:disabled,
.emotion-1:disabled + label {
  color: #304854;
}

.emotion-1,
.emotion-1:not(:placeholder-shown) {
  border-color: #ff4e64;
}

.emotion-1:focus-within {
  border-color: #8697a2;
}

<label
  className="emotion-1"
>
  <span>
    Choose file
  </span>
  <input
    className="emotion-0"
    type="file"
  />
</label>
`;
